In [1]:
import pandas as pd
import glob

In [2]:
path = r'C:\workspace\notebooks\health-ins-cross-sell\results'
all_files = glob.glob(path + "/*.csv")

In [3]:
summary_df = [pd.read_csv(f) for f in all_files]
summary_df_all=pd.concat(summary_df)
summary_df_all = summary_df_all.reset_index()
summary_df_all.drop('index',inplace=True,axis=1)

In [4]:
# total : Customer count treshold
# pct : Success percentage treshold
# col_count : Attribute complexity (1-4)
summary_df_all[(summary_df_all['total']>1000) & \
               (summary_df_all['pct']>.2)  & \
               (summary_df_all['col_count']==2) & \
               (summary_df_all['merged_columns_names'].apply(lambda x : 'Policy_Sales_Channel' not in x))  & \
               (summary_df_all['merged_columns_names'].apply(lambda x : 'Region_Code' not in x))]\
               .sort_values(by='pct',ascending=False)\
               .head(5)

merged_columns_names merged_columns_values        Y  \
3537175  Vehicle_Age >>> premium_group  > 2 Years >>> prem_4    514.0   
3537185      Vehicle_Age >>> age_group   > 2 Years >>> age_2   1339.0   
3537186      Vehicle_Age >>> age_group   > 2 Years >>> age_3   2254.0   
3538075   Vehicle_Damage >>> age_group         Yes >>> age_2  19813.0   
3537174  Vehicle_Age >>> premium_group  > 2 Years >>> prem_3   2271.0   

               N    total       pct  col_count  
3537175    933.0   1447.0  0.355418          2  
3537185   2532.0   3871.0  0.345985          2  
3537186   4645.0   6899.0  0.326764          2  
3538075  42328.0  62141.0  0.318845          2  
3537174   4900.0   7171.0  0.316743          2

## Notes


##### Quantitative Customer Personas for Health Insurance Cross Sell

* Average Cross sell rate is 12.4%

##### Indicators for increased likelyhood of cross sell
* Without previous insurance
* With Vehicle damage
* Between 32 and 46 years old
* Vehicle over 2 years old

##### Indicators for low likelyhood of cross sell
* No Vehicle damage
* Previously insured
* Vehicle below 1 year old
* 20 to 32 years old or 60 to 85 years old
* Female

##### Cross sell opportunities
* Customers between 32 and 46 years old with damaged vehicles (~31%)
* Customers between 32 and 46 years without previous insurance (~30%)
* Customers between 32 and 46 years without previous insurance with damaged vehicles (~33%)
* Customers between 32 and 46 with damaged vehicles paying premium between USD36292.0 and USD57361.0 (~33%)
* Customers between 32 and 60 with vehicles older than 2 years (~33%)
* Male customers with vehicles over 2 years old
* Customers with vechicles over 2 years old with premium above USD57361.0 (~35%)
* Customers between 32 to 46 years old with license but without previous insurance (~30%)
* Customers between 32 to 46 years old with damaged vehicles below 1 year old (~30%)